LibreNMS/Util/ObjectCache.php

```diff
@@ -0,0 +1,214 @@
+<?php
+/**
+ * ObjectCache.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2019 Tony Murray
+ * @author     Tony Murray <murraytony@gmail.com>
+ */
+
+namespace LibreNMS\Util;
+
+use App\Models\Application;
+use App\Models\BgpPeer;
+use App\Models\CefSwitching;
+use App\Models\Component;
+use App\Models\Device;
+use App\Models\OspfInstance;
+use App\Models\Port;
+use App\Models\Pseudowire;
+use App\Models\Sensor;
+use App\Models\Service;
+use App\Models\User;
+use App\Models\Vrf;
+use Cache;
+
+class ObjectCache
+{
+    private static $cache_time = 5;
+
+    public static function applications()
+    {
+        return Cache::remember('ObjectCache:applications_list', self::$cache_time, function () {
```

Comment:
Maybe?

```suggestion
        return Cache::remember('ObjectCache:applications_list' . auth()->user()->id, self::$cache_time, function () {
```

Reply:
Thank you @Jellyfrog 
Currently that is not an issue as this is only cached for the run duration (cache type array)  but it definitely should be.

Reply:
Fixed, thanks again :)